src/main/java/picard/sam/BamErrorMetric/BaseErrorCalculation.java

```diff
@@ -0,0 +1,203 @@
+                     final SAMLocusAndReferenceIterator.SAMLocusAndReference locusInfo);
+    }
+
+    /**
+     * An enum that is used to generate a {@link Supplier<BaseErrorCalculator>} from a string
+     * To use this given a String 'str':
+     * <p>
+     * Errors.valueOf(str).getErrorSupplier()
+     * <p>
+     * This is used in {@link CollectBamErrorMetrics} to convert an input argument to a {@link BaseErrorAggregation}.
+     */
+    enum Errors implements CommandLineParser.ClpEnum {
+        ERROR(BaseErrorCalculator::new, "Collects the average error at the bases provided."),
+        OVERLAPPING_ERROR(OverlappingReadsErrorCalculator::new, "Only considers bases from the overlapping parts of reads from the same template. " +
+                "For those bases, it calculates the error that can be attributable to pre-sequencing, versus during-sequencing.");
+
+        final private Supplier<? extends BaseCalculator> errorSupplier;
+
+        Errors(Supplier<? extends BaseCalculator> errorSupplier, final String docString) {
+            this.errorSupplier = errorSupplier;
+            this.docString = docString;
+        }
+
+        public Supplier<? extends BaseCalculator> getErrorSupplier() {
+            return errorSupplier;
+        }
+
+        private final String docString;
+
+        @Override
+        public String getHelpDoc() {
+            return docString + " Suffix is: " + errorSupplier.get().getSuffix();
+        }
+    }
+
+    /**
+     * A calculator that estimates the error rate of the bases it observes, assuming that the reference is truth.
+     */
+    public static class BaseErrorCalculator implements BaseCalculator {
+
+        long nMismatchingBases = 0;
+        long totalBases = 0;
+
+        @Override
+        public void addBase(final SamLocusIterator.RecordAndOffset recordAndOffset, final SAMLocusAndReferenceIterator.SAMLocusAndReference locusInfo) {
+            final byte readBase = recordAndOffset.getReadBase();
+            if (!SequenceUtil.isNoCall(readBase)) {
+                totalBases++;
+                if (readBase != locusInfo.referenceBase) {
+                    nMismatchingBases++;
+                }
+            }
+        }
+
+        @Override
+        public String getSuffix() {
+            return "error";
+        }
+
+        @Override
+        public ErrorMetrics.ComputableMetricBase getMetric() {
+            return new ErrorMetrics.SimpleErrorMetric("", totalBases, nMismatchingBases);
+        }
+    }
+
+    /**
+     * A calculator that estimates the error rate of the bases it observes, assuming that the reference is truth.
+     * This calculator only includes bases that have been read twice in the same template (once from each read) and
+     * thus only includes bases that arise from the overlapping part of the reads. Over those bases the Calculator
+     * distinguishes between weather the two reads agree with each other but differ from the reference (indicative
```

Comment:
typo: 
![image](https://user-images.githubusercontent.com/4700332/40743587-f5a63fa2-641f-11e8-8351-dff6d41e52e6.png)

Reply:
it raining typos!